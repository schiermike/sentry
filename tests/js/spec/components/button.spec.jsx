import React from 'react';
import {mount, shallow} from 'enzyme';
import Button from 'app/components/buttons/button';
import theme from 'app/utils/theme';

describe('Button', function() {
  it('renders', function() {
    let component = shallow(
      <Button theme={theme} priority="primary" size="large">
        Button
      </Button>
    );
    expect(component).toMatchSnapshot();
  });

  it('renders react-router link', function() {
    let component = shallow(<Button to="/some/route">Router Link</Button>);
    expect(component).toMatchSnapshot();
  });

  it('renders normal link', function() {
    let component = shallow(<Button href="/some/relative/url">Normal Link</Button>);
    expect(component).toMatchSnapshot();
  });

  it('renders disabled normal link', function() {
    let component = shallow(<Button href="/some/relative/url">Normal Link</Button>);
    expect(component).toMatchSnapshot();
  });

  it('calls `onClick` callback', function() {
    let spy = jest.fn();
    let component = mount(<Button theme={theme} onClick={spy} />);
    component.simulate('click');

    expect(spy).toHaveBeenCalled();
  });

  it('does not call `onClick` on disabled buttons', function() {
    let spy = jest.fn();
    let component = mount(<Button theme={theme} onClick={spy} disabled />);
    component.simulate('click');

    expect(spy).not.toHaveBeenCalled();
  });
});
