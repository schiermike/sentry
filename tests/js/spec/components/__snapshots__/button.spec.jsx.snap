// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders 1`] = `
<StyledSmartButton
  className=""
  disabled={false}
  onClick={[Function]}
  priority="primary"
  role="button"
  size="large"
  theme={
    Object {
      "alert": Object {
        "error": Object {
          "background": "#e03e2f",
          "backgroundLight": "#FDF6F5",
          "border": "#E7C0BC",
          "iconColor": "#C72516",
          "textDark": "#5d3e3b",
          "textLight": "#92635f",
        },
        "info": Object {
          "background": "#3B6ECC",
          "backgroundLight": "#F5FAFE",
          "border": "#B5D6ED",
          "iconColor": "#3B6ECC",
        },
        "success": Object {
          "background": "#57be8c",
          "backgroundLight": "#F8FCF7",
          "border": "#BBD6B3",
          "iconColor": "#3EA573",
        },
        "warn": Object {
          "background": "#ecc844",
          "backgroundLight": "#FFFDF7",
          "border": "#E1D697",
          "iconColor": "#e6bc23",
          "textDark": "#D3BE2B",
        },
        "warning": Object {
          "background": "#ecc844",
          "backgroundLight": "#FFFDF7",
          "border": "#E1D697",
          "iconColor": "#e6bc23",
          "textDark": "#D3BE2B",
        },
      },
      "blue": "#3B6ECC",
      "blueDark": "#2F58A3",
      "blueLight": "#628BD6",
      "borderDark": "#D1CAD8",
      "borderLight": "#E2DBE8",
      "borderLighter": "#f9f6fd",
      "borderRadius": "4px",
      "button": Object {
        "danger": Object {
          "background": "#e03e2f",
          "backgroundActive": "#bf2a1d",
          "border": "#bf2a1d",
          "borderActive": "#7d1c13",
          "color": "#FFFFFF",
        },
        "default": Object {
          "background": "#FFFFFF",
          "backgroundActive": "#FFFFFF",
          "border": "#d8d2de",
          "borderActive": "#c9c0d1",
          "color": "#2f2936",
          "colorActive": "#161319",
        },
        "disabled": Object {
          "background": "#FFFFFF",
          "backgroundActive": "#FFFFFF",
          "border": "#e3e5e6",
          "borderActive": "#e3e5e6",
          "color": "#ced3d6",
        },
        "link": Object {
          "background": "transparent",
          "backgroundActive": "transparent",
          "color": "#3B6ECC",
        },
        "primary": Object {
          "background": "#6C5FC7",
          "backgroundActive": "#4e3fb4",
          "border": "#3d328e",
          "borderActive": "#352b7b",
          "color": "#FFFFFF",
        },
        "success": Object {
          "background": "#3fa372",
          "backgroundActive": "#57be8c",
          "border": "#7ccca5",
          "borderActive": "#7ccca5",
          "color": "#FFFFFF",
        },
      },
      "dropShadowHeavy": "0 1px 4px 1px rgba(47,40,55,0.08), 0 4px 16px 0 rgba(47,40,55,0.12)",
      "dropShadowLight": "0 2px 0 rgba(37, 11, 54, 0.04)",
      "error": "#e03e2f",
      "fontSizeLarge": "18px",
      "fontSizeMedium": "16px",
      "fontSizeSmall": "12px",
      "gray1": "#BDB4C7",
      "gray2": "#9585A3",
      "gray3": "#645574",
      "gray4": "#4A3E56",
      "gray5": "#302839",
      "green": "#57be8c",
      "greenDark": "#3EA573",
      "greenLight": "#71D8A6",
      "greenTransparent": "rgba(87, 190, 140, 0.5)",
      "grid": 8,
      "offWhite": "#FAF9FB",
      "orange": "#ec5e44",
      "orangeDark": "#D3452B",
      "orangeLight": "#FF785E",
      "pink": "#F868BC",
      "pinkDark": "#DF4FA3",
      "pinkLight": "#FF82D6",
      "purple": "#6C5FC7",
      "purple2": "#6f617c",
      "purpleDark": "#5346AE",
      "purpleDarkest": "#392C94",
      "purpleLight": "#8679E1",
      "purplelightest": "#9F92FA",
      "red": "#e03e2f",
      "redDark": "#C72516",
      "redLight": "#FA5849",
      "settings": Object {
        "containerWidth": "1140px",
        "headerHeight": "115px",
        "sidebarWidth": "210px",
      },
      "success": "#57be8c",
      "text": Object {
        "family": "\\"Rubik\\", \\"Avenir Next\\", sans-serif",
        "familyMono": "Monaco, Consolas, \\"Courier New\\", monospace",
        "lineHeightBody": "1.4",
        "lineHeightHeading": "1.15",
      },
      "textColor": "#302839",
      "white": "#FFFFFF",
      "whiteDark": "#fbfbfc",
      "yellow": "#ecc844",
      "yellowDark": "#e6bc23",
      "yellowLight": "#FFF15E",
      "yellowLightest": "#FFFDF7",
      "yellowOrange": "#f9a66d",
      "yellowOrangeDark": "#E08D54",
      "yellowOrangeLight": "#FFC087",
      "zIndex": Object {
        "dropdown": 1001,
        "header": 1000,
        "modal": 10000,
        "sidebar": 100,
        "toast": 10001,
      },
    }
  }
>
  <ButtonLabel
    size="large"
  >
    Button
  </ButtonLabel>
</StyledSmartButton>
`;

exports[`Button renders disabled normal link 1`] = `
<StyledSmartButton
  className=""
  disabled={false}
  href="/some/relative/url"
  onClick={[Function]}
  role="button"
>
  <ButtonLabel>
    Normal Link
  </ButtonLabel>
</StyledSmartButton>
`;

exports[`Button renders normal link 1`] = `
<StyledSmartButton
  className=""
  disabled={false}
  href="/some/relative/url"
  onClick={[Function]}
  role="button"
>
  <ButtonLabel>
    Normal Link
  </ButtonLabel>
</StyledSmartButton>
`;

exports[`Button renders react-router link 1`] = `
<StyledSmartButton
  className=""
  disabled={false}
  onClick={[Function]}
  role="button"
  to="/some/route"
>
  <ButtonLabel>
    Router Link
  </ButtonLabel>
</StyledSmartButton>
`;
